web/src/main/java/org/mskcc/cbio/oncokb/api/pub/v1/TrialsApiController.java

```diff
@@ -0,0 +1,259 @@
+                            trials.add(curTrial);
+                        }
+                    }
+                }
+                result.put(SpecialTumorType.ALL_TUMORS.getTumorType(), trials);
+                return new ResponseEntity<>(result, status);
+            }
+
+            for (String cancerType: cancerTypes){
+                Set<String> nctIDSet = new HashSet<>();
+                List<Trial> addTrials = new ArrayList<>();
+                List<Trial> trials = new ArrayList<>();
+                if (SpecialTumorTypes.contains(cancerType)){
+                    trials = getTrialsForSpecialCancerTye(jsonObject, cancerType);
+                }
+                else{
+                    trials = getTrialsByCancerType(jsonObject, cancerType);
+                }
+                for (Trial trial: trials){
+                    if (!nctIDSet.contains(trial.getNctId())){
+                        nctIDSet.add(trial.getNctId());
+                        addTrials.add(trial);
+                    }
+                }
+                result.put(cancerType, addTrials);
+            }
+
+            if (result.isEmpty()){
+                status = HttpStatus.NOT_FOUND;
+                return new ResponseEntity<>(result, status);
+            }
+        }
+        return new ResponseEntity<>(result, status);
+    }
+
+    private Tumor getTumor(JSONObject jsonObject, String oncoTreeCode){
+        Tumor tumor = new Tumor();
+        if (jsonObject.containsKey(oncoTreeCode)){
+            JSONObject tumorObj = (JSONObject)jsonObject.get(oncoTreeCode);
+            Gson gson = new Gson();
+            tumor = gson.fromJson(tumorObj.toString(), Tumor.class);
+        }
+
+        return tumor;
+    }
+
+    private List<Trial> getTrialByTumorTreatment(List<Trial> trials, String treatment){
+        List<Trial> res = new ArrayList<>();
+        String[] tempDrugName = treatment.split(",|\\+");
+        for (int i = 0; i < tempDrugName.length; i++){
+            tempDrugName[i] = tempDrugName[i].trim();
+        }
+        Set<String> drugsNames = new HashSet<>(Arrays.asList(tempDrugName));
+        getTrialByTumorDrugName(res, trials, drugsNames);
+        return res;
+    }
+
+    private void getTrialByTumorDrugName(List<Trial> res, List<Trial> trials, Set<String> drugsNames){
+        for (Trial trial: trials){
+            for (Arms arm: trial.getArms()){
+                Set<String> armDrugNames = new HashSet<>();
+                for (Drug drug: arm.getDrugs()){
+                    armDrugNames.add(drug.getDrugName());
+                }
+                boolean isMatched = true;
+                for (String val: drugsNames){
+                    if (!armDrugNames.contains(val)){
+                        isMatched = false;
+                        break;
+                    }
+                }
+                if (isMatched){
+                    res.add(trial);
+                    break;
+                }

```

Comment:
I assume this is arm.getDrugs().stream().map(Drug::getDrugName).collect(Collectors.toSet()).containsAll(drugsNames)